src/__tests__/LabelGroup.test.tsx

```diff
@@ -1,28 +1,209 @@
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+    await user.click(expandButton)
+    await waitFor(() => getByLabelText('Close'))
+
+    expect(document.activeElement).toBe(getByLabelText('Close'))
+  })
+
+  it('should expand all tokens in place when overflowStyle="inline"', async () => {
+    const user = userEvent.setup()
+    const {getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3} overflowStyle="inline">
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+    const fourthLabel = getByText('Four')
+
+    expect(fourthLabel).not.toBeVisible()
+
+    await user.click(expandButton)
+
+    expect(fourthLabel).toBeVisible()
+  })
+
+  // TODO: test this some other way
+  // NOTE: It's not possible to test for side-effects of IntersectionObserver because it's a Jest mock
+  it.skip('should focus the first focusable token inline that was previously hidden when overflowStyle="inline"', async () => {
+    const user = userEvent.setup()
+    const {getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3} overflowStyle="inline">
+          <Token as="button" text="One" />
+          <Token as="button" text="Two" />
+          <Token as="button" text="Three" />
+          <Token as="button" text="Four" />
+          <Token as="button" text="Five" />
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+    const expandButton = getByText('+2')
+
+    expect(document.activeElement).not.toEqual(getByText('Four'))
+
+    await user.click(expandButton)
+    await waitFor(() => getByText('Four'))
+
+    expect(document.activeElement).toEqual(getByText('Four'))
+  })
+
+  it('should focus the collapse button when expanded, and the expand button when collapsed', async () => {
+    const user = userEvent.setup()
+    const {getByLabelText, getByText} = HTMLRender(
+      <ThemeAndStyleContainer>
+        <LabelGroup visibleTokenCount={3} overflowStyle="inline">
+          <Label>One</Label>
+          <Label>Two</Label>
+          <Label>Three</Label>
+          <Label>Four</Label>
+          <Label>Five</Label>
+        </LabelGroup>
+      </ThemeAndStyleContainer>,
+    )
+
+    // Expand
+    await user.click(getByText('+2'))
+    await waitFor(() => getByLabelText('Show less'))
+
+    expect(document.activeElement).toEqual(getByLabelText('Show less'))
+
+    // Collapse
+    await user.click(getByLabelText('Show less'))
+    await waitFor(() => getByText('+2'))
+
+    expect(document.activeElement?.textContent).toContain('+2')
```

Comment:
Help wanted: I had to write this weird assertion because `expect(document.activeElement).toEqual(getByText('+2'))` wasn't finding the element using `getByText('+2')`.